Sources/Logs.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Logs.swift
+//  swiftgen
+//
+//  Created by Olivier HALLIGON on 12/10/2017.
+//  Copyright © 2017 AliSoftware. All rights reserved.
+//
+
+import Foundation
+
+// MARK: Printing on stderr
+
+enum LogLevel {
+  case info, warning, error
+}
+
+enum ANSIColor: UInt8, CustomStringConvertible {
+  case reset = 0
+
+  case black = 30
+  case red
+  case green
+  case yellow
+  case blue
+  case magenta
+  case cyan
+  case white
+  case `default`
+
+  var description: String {
+    return "\u{001B}[\(self.rawValue)m"
+  }
+
+  func format(_ string: String) -> String {
+    if let termType = getenv("TERM"), String(cString: termType).lowercased() != "dumb" &&
+      isatty(fileno(stdout)) != 0 {
+      return "\(self)\(string)\(ANSIColor.reset)"
+    } else {
+      return string
+    }
+  }
+}
+
+func logMessage(_ level: LogLevel, _ string: CustomStringConvertible) {
```

Comment:
It might be nicer to have a logger system that accepts a logging level (info or above, warning or above, etc...) and filters out (does not log) messages below the current log level. When we switch on "verbose" logging, we'd set the logger's level to `info` instead of `warning`.
- No longer need to check at the log point if it should be logged
- Introduce even more verbose logging (-vvv) at some point
- Introduce a silent mode at some point (disables all logging)
- Have the plugins (swiftgenkit) accept the logger to do their own logging if needed

Might be worth it checking some of the minimal existing logging libraries for ideas (and maybe dependency).

Reply:
Also, if we decide to do this, the `_ string` message should be lazily evaluated to avoid processing expensing messages.

Reply:
Def a good idea for later, I agree (but I wanna do the big merge first, otherwise we'll forever postpone it)